lib/Parse/ParseExpr.cpp

```diff
@@ -2040,7 +2040,9 @@ ParserResult<Expr> Parser::parseExprStringLiteral() {
     // so that parseDeclPoundDiagnostic() can figure out why this string
     // literal was bad.
     return makeParserErrorResult(
-        new (Context) InterpolatedStringLiteralExpr(Loc, 0, 0, nullptr));
+        new (Context) InterpolatedStringLiteralExpr(Loc,
+                                                    EndLoc.getAdvancedLoc(-1),
```

Comment:
Hmm...we do have cases where the trailing quote is not one character. Way back towards the beginning of this function, we compute a `CloseQuoteBegin` offset from the `StartLoc` for this (it was added pretty recently; might not have been there when you started). That would probably be better.

Reply:
Thanks! Will fix.